docs/decisions/0017-generalized-containers.rst

```diff
@@ -0,0 +1,79 @@
+- Containers will follow extensibility principles in `Content Extensibility Through Model Relations <0003-content-extensibility.rst>`_ for creating new container types or subtypes.
+
+3. Container Children and Relationships
+=======================================
+
+This section defines container children, their order, and relationships, covering flexible connections and support for draft and published states of their children.
+
+- Each container version holds a list of children that the author has defined for that version.
+- The author-defined list is used to show the content of a container version as the author specified it.
+- The author-defined list won't change for a specific container version even if its members change. E.g., a unit version UV1 with three components (CV1, CV2, CV3) will always have those three components in the author-defined list, even if one of the components is soft-deleted or a new version for the component is created.
+- The children of a container can be any type of publishable content. E.g., sections, subsections, units, components, and any other publishable thing. For more details on publishable content, see `PublishableEntity`_.
+- Children within a container are maintained in a specific order as an ordered list. E.g., components within a unit, or units within a subsection, are presented in a specific order.
+- Containers represent their content hierarchy through a structure, like Course > Section > Subsection > Unit > Component, which defines parent-child relationships at each level.
+- Containers can reference a specific version of their children or be set to point to their latest versions. For instance, component V1 might be used in a unit instead of its latest version. The latest version of a child can be referenced by setting its version to ``None`` which consists of the chosen standard for this representation.
+- A single child (publishable entity) can be shared by multiple containers, allowing for reuse of content across different containers. For instance, a component can be shared by multiple units.
+
+4. Next Container Versions
+==========================
+
+This section defines the rules for version control in containers, explaining when new versions are created based on changes to container structure or metadata.
+
+- A new version is created if and only if the container itself changes (e.g., title, ordering of children, adding or removing children) and not when its children change (e.g., a component in a Unit is updated with new text). For instance, a new version of a unit is created when a component is removed, not when a new version of a component is created.
```

Comment:
> a new version of a unit is [not created] when a new version of a component is created.

How does this interact with the library workflows?

You have a unit in a library with one component. You publish everything and use it in a course. You edit that component in the library. The unit does not show as changed and can't be published. The component _does_ show as changed, and can be published. When you publish it, you'll see a notification in the course that a newer version of the component is available, and you'll update it directly, not via updating the unit?

Reply:
I imagine that would happen if we set the unit component to point to the latest version. We wouldn't need to update the unit itself because nothing really changed. Does that make sense?

Reply:
Yeah, I think I understand this now.

> You have a unit in a library with one component. You publish everything and use it in a course. You edit that component in the library. The unit ~~does not show as changed and can't be published~~ **shows that it _contains_ unpublished changes and you can publish those changes** ~~The component does show as changed, and can be published.~~ When you publish it, you'll see a notification in the course that a newer version of the component is available, and you'll update it directly, not via updating the unit ~~?~~.